src/it/java/org/checkstyle/plugins/sonar/RunPluginTest.java

```diff
@@ -0,0 +1,163 @@
+package org.checkstyle.plugins.sonar;
+
+import static org.junit.Assert.fail;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import com.sonar.orchestrator.Orchestrator;
+import com.sonar.orchestrator.OrchestratorBuilder;
+import com.sonar.orchestrator.build.Build;
+import com.sonar.orchestrator.build.BuildResult;
+import com.sonar.orchestrator.build.MavenBuild;
+import com.sonar.orchestrator.build.SonarScanner;
+import com.sonar.orchestrator.locator.FileLocation;
+import com.sonar.orchestrator.locator.Location;
+import com.sonar.orchestrator.locator.MavenLocation;
+import com.sonar.orchestrator.locator.URLLocation;
+import org.sonarqube.ws.WsMeasures;
+import org.sonarqube.ws.client.HttpConnector;
+import org.sonarqube.ws.client.WsClient;
+import org.sonarqube.ws.client.WsClientFactories;
+import org.sonarqube.ws.client.measure.ComponentWsRequest;
+
+/**
+ * a lot of code from https://github.com/SonarSource/sonar-java/blob/5aaaeba90cd85a40391706ad2f806d6a156f80cb/its/ruling/src/test/java/org/sonar/java/it/JavaRulingTest.java
```

Comment:
we do not need such comment any more, I used it as placeholder  for links where I found good examples.

Reply:
removed